API/jsi/jsi/jsi.h

```diff
@@ -267,6 +270,12 @@ class JSI_EXPORT Runtime {
   /// which returns no metrics.
   virtual Instrumentation& instrumentation();
 
+  /// Creates a Node-API environment.
+  /// \throw a \c JSINativeException if the runtime does not support Node-API.
+  /// \param napi_env the output parameter used to store the newly created
+  /// Node-API environment.
+  virtual void createNodeApiEnv(int32_t apiVersion, napi_env *);
```

Comment:
I guess for the complete API will also need to add the `const char* moduleFileName` parameter as in the Node.js code:
https://github.com/nodejs/node/blob/5a2614fd03b93da972c34c4842d3cb6bd12be65a/src/node_api_internals.h#L12
```C++
  node_napi_env__(v8::Local<v8::Context> context,
                  const std::string& module_filename,
                  int32_t module_api_version);
```

Reply:
We definitely could. Would that be useful for anything? 🤔 Perhaps serve as an id of the module across `napi_env`s?

Reply:
@vmoroz any thoughts on this before the PR is merged? We can certainly also revisit this later. My immediate thought is that we keep it out to keep things simple if the implementation doesn't have any use of the parameter anyway 🤔 But I'd be happy to change my mind on that 🙂